pymc3/tests/test_distributions_random.py

```diff
@@ -251,6 +251,14 @@ def test_different_shapes_and_sample_sizes(self, shape):
                 assert e == a
 
 
+class TestGaussianRandomWalk(BaseTestCases.BaseTestCase):
+    distribution = pm.GaussianRandomWalk
+    params = {'mu': 1., 'sigma': 1.}
+
+    @pytest.mark.xfail(reason="Supporting this makes a nasty API")
```

Comment:
Yes, i tried writing a random method for the time series but there are so many shape issues that can come up that I stopped. However, I feel that an important test is to check that `sample_prior_predictive` doesn't fail if `mu` or `sigma` are scalars RVs. `cumsum(..., axis=0)` could be wrong there because of the `size` prepend.

Reply:
I think the existing test checks this for scalar rvs, but not like, broadcasting ones. This is the stochastic volatility example that I wanted to get working (and now it works...)

![image](https://user-images.githubusercontent.com/2295568/69014972-6fc4d300-095d-11ea-88dc-8153f8c2b55b.png)

Reply:
Let me know if you have a safer alternative. My brief notes on this are:

Given 
```python
rv = st.norm(np.ones(10), 1)
```
Calling
```python
rv.rvs(size)
```
fails unless `size` is a tuple that ends with a 10 (or is `None`). So `size=10`, `size=(20, 10)`, `size=(20, 2, 10)` all work, but `size=(10, 20)` fails. 

I was sort of hoping that , by the magic you + rpgoldman did in the sampling code that `size` would already have the right shape to respect that (in particular, it throws useful errors if I return something with a bad shape!) If you have some directions I might go in to make this more robust, I can follow that, but I was hoping to just cover some basic functionality with straightforward code for now.

Reply:
uh... this comment was supposed to go with the `axis=0` comment you made. apologies!

Reply:
The shape stuff I did in `generate_samples` should already give a `size` that matches with `mu` and `sigma`'s shapes. This `size` isn't the `sample_shape` passed to `random`, but something compatible with scipy's `size`.
To get the `sample_shape` parameter in `_random` you could do this:

```python
def _random(self, sigma, mu, size, sample_shape):
    if size[:len(sample_shape)] == sample_shape:
        axis = len(sample_shape)
    else:
        axis = 0
....
```

and in the call to `generate_samples` you should pass the parameter `not_broadcast_kwargs={"sample_shape": to_tuple(size)}`. This will forward the inputted `size` to `_random`

Reply:
Thanks again -- you're right, and I finally recognized the bug that you caught :smile: 

I updated with your suggestions, *and* required a timeseries to have a positive shape supplied, so that we can sum along this timeseries axis.